src/config.c

```diff
@@ -3439,6 +3439,55 @@ void addModuleNumericConfig(sds name, sds alias, int flags, void *privdata, long
     }
 }
 
+int moduleServerConfigSet(client *c, sds name, sds value) {
+    standardConfig *config = lookupConfig(name);
+
+    if (!config) return 0;
+
+    if (config->flags & MODULE_CONFIG) return 0;
+
+    if (config->flags & IMMUTABLE_CONFIG ||
+        (config->flags & PROTECTED_CONFIG &&
+         !allowProtectedAction(server.enable_protected_configs, c)))
+        return 0;
+
+    if (server.loading && config->flags & DENY_LOADING_CONFIG) return 0;
+
+    sds old_value = config->interface.get(config);
+    const char *errstr = NULL;
+    int res = performInterfaceSet(config, value, &errstr);
+    if (!res) {
+        serverLog(LL_WARNING, "Failed setting new configuration from module. Error setting %s: %s", config->name, errstr);
+        goto err;
+    }
+    if (res != 1) goto end; /* VOLATILE_CONFIG */
```

Comment:
this comment is somewhat confused. when res is 2, which means that the config hasn't been changed, but it doesn't mean that this config is volatile config.
if config is volatile, res still returns 1 even if these is no change.